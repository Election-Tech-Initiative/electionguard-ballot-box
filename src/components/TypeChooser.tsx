import React, { useContext } from 'react'
import styled from 'styled-components'
import BallotBoxContext from '../contexts/ballotBoxContext'
import { BallotBoxType } from '../config/types'
import Button from './Button'
import { Wobble } from './Animations'

const Line = styled.hr`
  width: 100%;
`

const Container = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
`

const ButtonOptions = styled.div`
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: center;
  > * {
    margin: 1rem;
  }
`

const TypeChooser = () => {
  const { setType } = useContext(BallotBoxContext)
  return (
    <Container>
      <h1>Ballot Box</h1>
      <Line />
      <p>Select Ballot Box Type</p>
      <ButtonOptions>
        <Wobble>
          <Button
            primary
            big
            onPress={() => setType(BallotBoxType.Cast)}
            aria-label="Select cast as ballot box type"
          >
            {BallotBoxType[BallotBoxType.Cast]}
          </Button>
        </Wobble>
        <Wobble>
          <Button
            primary
            big
            onPress={() => setType(BallotBoxType.Spoil)}
            aria-label="Select spoil as ballot box type"
          >
            {BallotBoxType[BallotBoxType.Spoil]}
          </Button>
        </Wobble>
      </ButtonOptions>
    </Container>
  )
}

export default TypeChooser
