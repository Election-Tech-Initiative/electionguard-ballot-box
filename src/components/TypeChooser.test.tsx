import React from 'react'

import { render } from '../../test/testUtils'

import TypeChooser from './TypeChooser'

it('can renders TypeChooser', () => {
  const { container } = render(<TypeChooser />, {})
  expect(container.firstChild).toMatchSnapshot()
})
