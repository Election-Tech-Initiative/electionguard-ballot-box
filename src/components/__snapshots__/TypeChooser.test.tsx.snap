// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`can renders TypeChooser 1`] = `
<div
  class="sc-bZQynM bsmBBS"
>
  <h1>
    Ballot Box
  </h1>
  <hr
    class="sc-EHOje YINzU"
  />
  <p>
    Select Ballot Box Type
  </p>
  <div
    class="sc-gzVnrw cOUDlP"
  >
    <div
      class="sc-bxivhb woZhC"
    >
      <button
        aria-label="Select cast as ballot box type"
        class="sc-bwzfXH eHmtkf"
        type="button"
      >
        Cast
      </button>
    </div>
    <div
      class="sc-bxivhb woZhC"
    >
      <button
        aria-label="Select spoil as ballot box type"
        class="sc-bwzfXH eHmtkf"
        type="button"
      >
        Spoil
      </button>
    </div>
  </div>
</div>
`;
